src/test/java/StringManipulationTest.java

```diff
@@ -55,27 +61,33 @@ public void testRemoveNthCharacter2() {
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello World");
+        //   out of bounds error   --> NOTE: this should fail (because n > 11) but it does not.  Realized this too late
+        manipulatedstring.removeNthCharacter(20, false);
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello World is such an overused phrase:)");
+        assertEquals("Hello Wor d is such an overus d phrase: ", manipulatedstring.removeNthCharacter(10, true));
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello     --5blank spaces--     World");
+        assertEquals("Helo   -5ban sacs-    ord", manipulatedstring.removeNthCharacter(3, false));
     }
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello World");
+        assertEquals("           ", manipulatedstring.removeNthCharacter(1, true));
```

Comment:
This is a smart test to make, no issues, just seeing if retains the space that we told it to keep.